kgforge/specializations/models/rdf/service.py

```diff
@@ -162,18 +181,20 @@ def validate(self, resource: Resource, type_: str):
                     "Resource has list of types as attribute and type_ parameter is not specified. "
                     "Please provide a type_ parameter to validate against it."
                 )
-            if type_ is None:
-                shape_iri = self.types_to_shapes[resource.type]
-            else:
-                shape_iri = self.types_to_shapes[type_]
+            type_to_validate = type_ if type_ else resource.type
+            shape_iri = self.get_shape_iri_from_class_fragment(type_to_validate)
         except AttributeError as exc:
-            raise TypeError("Resource requires a type attribute") from exc
-
+            raise TypeError(
+                "Resource requires a type attribute or a type_ parameter should be provided"
+            ) from exc
         data_graph = as_graph(resource, False, self.context, None, None)
-        return self._validate(shape_iri, data_graph)
+        shape, shacl_graph = self.get_shape_graph(shape_iri)
+        return self._validate(shape_iri, data_graph, shape, shacl_graph)
 
     @abstractmethod
-    def _validate(self, iri: str, data_graph: Graph) -> Tuple[bool, Graph, str]:
+    def _validate(
+        self, iri: str, data_graph: Graph, shape: Shape, shacl_graph: Graph
```

Comment:
Both implementations (StoreSrrvice and DirectoryService) do not have shacl_graph as an extra parameter.